homeassistant/components/smappee/binary_sensor.py

```diff
@@ -0,0 +1,192 @@
+        """Return the unique ID for this binary sensor."""
+        return (
+            f"{self._service_location.device_serial_number}-"
+            f"{self._service_location.service_location_id}-"
+            f"presence"
+        )
+
+    @property
+    def device_info(self):
+        """Return the device info for this binary sensor."""
+        return {
+            "identifiers": {(DOMAIN, self._service_location.device_serial_number)},
+            "name": self._service_location.service_location_name,
+            "manufacturer": "Smappee",
+            "model": self._service_location.device_model,
+            "sw_version": self._service_location.firmware_version,
+            "via_device": (DOMAIN, self._smappee_base.smappee.username),
+        }
+
+    async def async_update(self):
+        """Get the latest data from Smappee and update the state."""
+        await self._smappee_base.async_update()
+
+        self._state = self._service_location.is_present
+
+
+class SmappeeAppliance(BinarySensorEntity):
+    """Implementation of a Smappee binary sensor."""
+
+    def __init__(
+        self,
+        smappee_base,
+        service_location,
+        appliance_id,
+        appliance_name,
+        appliance_type,
+    ):
+        """Initialize the Smappee sensor."""
+        self._smappee_base = smappee_base
+        self._service_location = service_location
+        self._appliance_id = appliance_id
+        self._appliance_name = appliance_name
+        self._appliance_type = appliance_type
+        self._state = False
+        self._appliance_power = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return (
+            f"{self._service_location.service_location_name} - "
+            f"{BINARY_SENSOR_PREFIX} - "
+            f"{self._appliance_name if self._appliance_name != '' else self._appliance_type}"
+        )
+
+    @property
+    def is_on(self):
+        """Return if the binary sensor is turned on."""
+        return self._state
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return STATE_ON if self._state else STATE_OFF
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        icon_mapping = {
+            "Freezer": "mdi:fridge",
+            "Refrigerator": "mdi:fridge",
+            "Coffeemaker": "mdi:coffee",
+            "Washing Machine": "mdi:washing-machine",
+            "Clothes Dryer": "mdi:tumble-dryer",
+            "Dishwasher": "mdi:dishwasher",
+            "Clothes Iron": "mdi:hanger",
+            "Stove": "mdi:stove",
+            "Oven": "mdi:stove",
+            "Microwave": "mdi:microwave",
+            "Fan": "mdi:fan",
+            "Car Charger": "mdi:car",
+            "Water Pump": "mdi:water-pump",
+        }
+        return (
+            icon_mapping.get(self._appliance_type)
+            if self._appliance_type in icon_mapping
+            else "mdi:power-plug"
```

Comment:
What does the power plug represent? Can we set a device class instead?

https://developers.home-assistant.io/docs/core/entity/binary-sensor#available-device-classes

Reply:
The power plug represents an electrical appliance.

Reply:
We could use the `"plug"` device class?

Reply:
But then we will need to remove the icon selection again?

Reply:
The plug appliance is missing in the icon mapping so it's easy to return None for the icon in that case.